src/qutip_tensornetwork/core/data/tensor_train/tensor_train.py

```
@@ -234,6 +234,119 @@ def _to_tt_format(self):
 
         self._nodes = set(nodes)
 
+    def truncate(self, bond_dimension=None, max_truncation_err=None):
+        """Truncate in-place the bond dimension of the tensor train according
+        to ``bond_dimension``. This is done from left to right and once
+        finished the network is right normalized.
+
+        Notes
+        -----
+        The truncation method consists on performing a svd decomposition from
+        left to right and truncating each nodes svd by the given parameters.
+
+        Parameters
+        ----------
+        bond_dimension: List of int or int
+            List of integers that define, from left to right, the target bond
+            dimension. If a single integer is provided is understood as the
+            same bond dimension for all bonds. None is understood as
+            "infinite" bond dimension and hence no truncation is done.
+
+        max_truncation_err: List of float or float
+            Maximum truncation error for each individual node. If a single
+            value is provided instead of a list, it is the assumed that the
+            same value applies for every node.
+
+        Returns
+        -------
+        error: float
+            List of truncated singular values.
+
+        See also
+        --------
+        tensornetwork.split_node:
+            Function employed to split individual nodes. ``bond_dimension``
+            and ``max_truncation_err`` directly translate into
+            ``max_singular_values`` and ``max_truncation_err``.
+        """
+        if not isinstance(bond_dimension, list):
+            bond_dimension = [bond_dimension] * len(self.bond_edges)
+
+        if not isinstance(max_truncation_err, list):
+            max_truncation_err = [max_truncation_err] * len(self.bond_edges)
+
```

We should check that `len(bond_dimension) == len(self.bond_edges)` and `len(max_truncation_err) == len(self.bond_edges)` and raise a ValueError if they are not (since otherwise `zip(self.bond_edges, ...)` will silently drop any extra elements and create a mess.